ATC crashes due to a bug of Lidar

## Bug Report

I have upgraded  two clusters to 5.7.0 last Monday with Lidar enabled. Today we noticed that some ATCs crashed a few times.  Trace stack:

```
panic: value method github.com/concourse/concourse/vars.UndefinedVarsError.Error called using nil *UndefinedVarsError pointer
goroutine 65423631 [running]:
github.com/concourse/concourse/vars.(*UndefinedVarsError).Error(0x0, 0xc0079e54c0, 0x1f1e166)
	<autogenerated>:1 +0x71
github.com/concourse/concourse/atc/db.(*resource).SetCheckSetupError(0xc002e55a20, 0x2830840, 0x0, 0x4218a5, 0x1dcb260)
	/tmp/build/1c3187db/concourse/atc/db/resource.go:280 +0x130
github.com/concourse/concourse/atc/lidar.(*scanner).setCheckError(0xc000c5b7c0, 0x2895fa0, 0xc00079d920, 0x28aad00, 0xc002e55a20, 0x2830840, 0x0)
	/tmp/build/1c3187db/concourse/atc/lidar/scanner.go:119 +0x4c
github.com/concourse/concourse/atc/lidar.(*scanner).Run.func1(0xc0021d43a0, 0xc000c5b7c0, 0xc00686ed20, 0x28c4a00, 0xc002e55a20, 0xc0025ba000, 0xbff, 0xe00)
	/tmp/build/1c3187db/concourse/atc/lidar/scanner.go:64 +0x189
created by github.com/concourse/concourse/atc/lidar.(*scanner).Run
	/tmp/build/1c3187db/concourse/atc/lidar/scanner.go:60 +0x2d9
```

## Steps to Reproduce

Not sure, this bug should be caused by some race condition.

Basically, needs to enable lidar, and make some resource check fails.

## Expected Results

ATC should never crash.

## Actual Results

It crashed intermittently.

## Additional Context

> (delete this section if not needed)

## Version Info

* Concourse version: 5.7.0, but I believe this bug has been there since Lidar was introduced
* Deployment type (BOSH/Docker/binary): Docker
* Infrastructure/IaaS: K8S
* Browser (if applicable): NA
* Did this used to work? No

--- a/atc/lidar/scanner.go
+++ b/atc/lidar/scanner.go
@@ -60,7 +60,7 @@ func (s *scanner) Run(ctx context.Context) error {
 		go func(resource db.Resource, resourceTypes db.ResourceTypes) {
 			defer waitGroup.Done()
 
-			err = s.check(resource, resourceTypes)
+			err := s.check(resource, resourceTypes)
 			s.setCheckError(s.logger, resource, err)
 
 		}(resource, resourceTypes)

